data/src/main/java/com/linkedin/util/DataComplexUtil.java

```diff
@@ -0,0 +1,187 @@
+package com.linkedin.util;
+
+import com.linkedin.data.Data;
+import com.linkedin.data.DataList;
+import com.linkedin.data.DataMap;
+import com.linkedin.data.DataMapBuilder;
+import com.linkedin.data.collections.CheckedUtil;
+import com.linkedin.data.template.DataTemplateUtil;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Utils to convert generic List/Map to DataList/DataMap or vice versa
+ */
+public class DataComplexUtil {
+
+  private DataComplexUtil() { }
+
+  /**
+   * Converts the input {@link Map} to a {@link DataMap} recursively.
+   * Optional to stringfy the value of the leaf nodes in the datacomplex
+   */
+  public static DataMap convertMap(Map<String, ?> input, boolean stringify) {
```

Comment:
It's annoying but can follow the Restli repo indentation ie '{' in a new line

Reply:
yeah it is call AllMan's style. https://en.wikipedia.org/wiki/Indentation_style

Java code doesn't usually do that though